File: simple_linear_regression_temp.py
#!/usr/bin/env python3
# -*- coding: utf-8 -*-
"""
Created on Thu Aug  3 22:51:42 2017

@author: AaronNguyen
"""

import numpy as np
import matplotlib.pyplot as plt
import pandas as pd

# Importing the dataset
dataset = pd.read_csv('Salary_Data.csv')
X = dataset.iloc[:, :-1].values
y = dataset.iloc[:, 1].values

# Splitting the dataset into the Training set and Test set
from sklearn.cross_validation import train_test_split
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size = 1/3, random_state = 0)

# Feature Scaling
"""from sklearn.preprocessing import StandardScaler
sc_X = StandardScaler()
X_train = sc_X.fit_transform(X_train)
X_test = sc_X.transform(X_test)
sc_y = StandardScaler()
y_train = sc_y.fit_transform(y_train)"""

# Fitting the simple linear regression to the training set
from sklearn.linear_model import LinearRegression
regressor = LinearRegression()
regressor.fit(X_train,y_train) # already learn the correlation from training set

# Predicting the Test set results
y_pred = regressor.predict(X_test)  # vector of prediction of dependent variable

# Visual the training test results
plt.scatter(X_train, y_train,color = 'red') # (x coordinate,y coordinate)
plt.plot(X_train,regressor.predict(X_train), color = 'blue')
plt.title('Salary vs Experience (Training set)')
plt.xlabel('Years of Experience')
plt.ylabel('Salary')
plt.show()

# Visual the Test set results
plt.scatter(X_test, y_test,color = 'red') # (x coordinate,y coordinate)
plt.plot(X_train,regressor.predict(X_train), color = 'blue')
plt.title('Salary vs Experience (Test set)')
plt.xlabel('Years of Experience')
plt.ylabel('Salary')
plt.show()

File: simple_linear_regression_temp.R
# Simple Linear Regression

# Importing the dataset
dataset = read.csv('Salary_Data.csv')

# Splitting the dataset into the Training set and Test set
# install.packages('caTools')
library(caTools)
set.seed(123)
split = sample.split(dataset$Salary, SplitRatio = 2/3)
training_set = subset(dataset, split == TRUE)
test_set = subset(dataset, split == FALSE)

# Feature Scaling
# training_set = scale(training_set)
# test_set = scale(test_set)

# Fitting Simple Linear Regression to the Training set
regressor = lm(formula = Salary ~ YearsExperience,
               data = training_set) # dependant variable ~ independant variable

# P value < 5% means the independent variable has significant effectssu

#Predict the Test set results
y_pred = predict(regressor, newdata = test_set)

# Visual of Training and Test set observations
# install.packages('ggplot2')
#library(ggplot2)

#First plot all observations points , start with trainng set
ggplot() + 
  geom_point(aes (x = training_set$YearsExperience, y = training_set$Salary),
             colour = 'red') +
  geom_line(aes(x = training_set$YearsExperience,predict(regressor, newdata = training_set)),
            colour = 'blue') +
  ggtitle('Salary vs Experience (Traing set)') + 
  xlab('Years of experience') +
  ylab('Salary')

# Test set
ggplot() + 
  geom_point(aes (x = test_set$YearsExperience, y = test_set$Salary),
             colour = 'red') +
  geom_line(aes(x = training_set$YearsExperience,predict(regressor, newdata = training_set)),
            colour = 'blue') +
  ggtitle('Salary vs Experience (Test set)') + 
  xlab('Years of experience') +
  ylab('Salary')